### DISCUSSION TITLE
Transaction Rejected after chain switch

### DISCUSSION THREAD

**Discussion by JaySecurity** [MetaMask/metamask-sdk#1115] [Q&A]
When making a purchase a user has options of a token on ethereum or base.  As part of the purchase logic the current chain is checked and if required eth_switchEthereumChain is called. in the case when this request is made after completing the switch metamask mobile app closes and the eth_sendTransaction gets called and rejected without even opening metamask. The same behavior on desktop works fine. 

Using  "@metamask/sdk-react": "^0.30.0", Brave and Firefox mobile browsers and metamask mobile app on android, all updated to latest versions.

Any thoughts on what might be causing this would be appreciated.

Cheers

**Comment by christopherferreira9** [comment]
Hi @JaySecurity ! 
Are you able to share a part of the code for that interaction? 
For these sorts of interactions we suggest the use of `metamask_batch` where the first request is the `wallet_switchEthereumChain` and the second is the `eth_sendTransaction`. The call for `eth_sendTransaction` should not be programatically made to avoid having the deeplink blocked by the mobile OS.

**Reply by JaySecurity** [reply]
I'll try to get some code available to share and I'll have a look at the batch option. I've had to jump over to another project for a few day but I'll update as soon as I get a chance. Thanks for the response.
